core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -462,9 +463,18 @@ public ViewMetadata build() {
           ViewProperties.VERSION_HISTORY_SIZE,
           historySize);
 
-      // expire old versions, but keep at least the versions added in this builder
-      int numAddedVersions = (int) changes(MetadataUpdate.AddViewVersion.class).count();
-      int numVersionsToKeep = Math.max(numAddedVersions, historySize);
+      // expire old versions, but keep at least the versions added in this builder and the current
+      // version
+      int numVersions =
+          ImmutableSet.builder()
+              .addAll(
+                  changes(MetadataUpdate.AddViewVersion.class)
+                      .map(v -> v.viewVersion().versionId())
+                      .collect(Collectors.toSet()))
+              .add(currentVersionId)
+              .build()
+              .size();
```

Comment:
Minor: Isn't this the same as 

```
(int) changes(MetadataUpdate.AddViewVersion.class).count() + 1; 
```

Feels a bit easier to read that way along with the comment since I don't think we actually need the set.

Reply:
Yes, I also think we need a set here in here. My original comment was to avoid introducing 2 sets as the code was doing so, one is enough instead.